src/apps/eth/eth_verify.c

```diff
@@ -115,23 +122,39 @@ app_eth_sign_error_t app_eth_verify_erc20_transaction(const ETHSignRequest* requ
     }
     bignum256 value_scalar;
     _bigendian_to_scalar(value, 32, &value_scalar);
-    const _amount_t amount = {
-        .unit = erc20_params->unit,
-        .decimals = erc20_params->decimals,
-        .value = &value_scalar,
-    };
-    app_eth_sign_error_t result = _verify_recipient(recipient, &amount);
-    if (result != APP_ETH_SIGN_OK) {
-        return result;
-    }
-    const _amount_t total = {
-        .unit = erc20_params->unit,
-        .decimals = erc20_params->decimals,
-        .value = &value_scalar,
-    };
-    result = _verify_total_fee(request, &total, params->unit);
-    if (result != APP_ETH_SIGN_OK) {
-        return result;
+    if (erc20_params != NULL) {
+        const _amount_t amount = {
+            .unit = erc20_params->unit,
+            .decimals = erc20_params->decimals,
+            .value = &value_scalar,
+        };
+        app_eth_sign_error_t result = _verify_recipient(recipient, &amount);
+        if (result != APP_ETH_SIGN_OK) {
+            return result;
+        }
+        const _amount_t total = {
+            .unit = erc20_params->unit,
+            .decimals = erc20_params->decimals,
+            .value = &value_scalar,
+        };
+        result = _verify_total_fee(request, &total, params->unit);
+        if (result != APP_ETH_SIGN_OK) {
+            return result;
+        }
+    } else {
+        app_eth_sign_error_t result = _verify_recipient(recipient, NULL);
+        if (result != APP_ETH_SIGN_OK) {
+            return result;
+        }
+        const _amount_t total = {
+            .unit = NULL,
+            .decimals = 0,
+            .value = &value_scalar,
+        };
+        result = _verify_total_fee(request, &total, params->unit);
```

Comment:
pass `NULL` instead of `&total`, as you do not need any of the fields, not even value